src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(value) {
+      return (this.result += value);
+    },
+
+    subtract(value) {
+      return (this.result -= value);
+    },
+
+    multiply(value) {
+      return (this.result *= value);
+    },
+
+    divide(value) {
+      if (value === 0) {
+        return undefined;
+      }
+
+      return (this.result /= value);
+    },
+
+    operate(callback, value) {
+      callback.call(this, value);
```

Comment:
[CODE KNOWLEDGE] - It's good that you're using `call` to ensure the correct `this` context, but you should also check if `callback` is a function before attempting to call it. This can prevent potential runtime errors if a non-function is passed.